deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.kt

```diff
@@ -0,0 +1,580 @@
+                allDeepLinkAnnotatedElements,
+                annotatedClassElements,
+                annotatedMethodElements
+            )
+
+            // Create DeepLinkDelegate classes
+            createDeeplinkDelegates(roundEnv = round)
+
+            // Create DeepLinkRegistry classes
+            createDeeplinkRegistries(
+                roundEnv = round,
+                annotatedClassElements = annotatedClassElements,
+                annotatedMethodElements = annotatedMethodElements,
+                deepLinkElements = collectDeepLinkElements(
+                    prefixes = customAnnotationPrefixes(customAnnotations),
+                    classElementsToProcess = annotatedClassElements,
+                    methodElementsToProcess = annotatedMethodElements
+                ),
+            )
+        } catch (e: DeepLinkProcessorException) {
+            logError(element = e.element,
+                    message = e.message ?: "")
+        }
+    }
+
+    private fun collectDeepLinkElements(
+            prefixes: Map<XType, Array<String>>,
+            classElementsToProcess: Set<XTypeElement>,
+            methodElementsToProcess: Set<XMethodElement>
+    ): List<DeepLinkAnnotatedElement> {
+        return (
+                classElementsToProcess.map { element ->
+                    mapUrisToDeepLinkAnnotatedElement(element, prefixes)
+                }.flatten() + methodElementsToProcess.map { element ->
+                    verifyMethod(element)
+                    mapUrisToDeepLinkAnnotatedElement(element, prefixes)
+                }.flatten()
+                ).filterNotNull()
+    }
+
+    private fun mapUrisToDeepLinkAnnotatedElement(
+            element: XElement,
+            prefixes: Map<XType, Array<String>>
+    ): List<DeepLinkAnnotatedElement?> = getAllUrisForAnnotatedElement(element, prefixes).map { uri ->
+        try {
+            DeepLinkAnnotatedElement(uri, element)
+        } catch (e: MalformedURLException) {
+            environment.messager.printMessage(
+                    kind = Diagnostic.Kind.ERROR,
+                    msg = "Malformed Deep Link URL $uri"
+            )
+            null
+        }
+    }
+
+    private fun getAllUrisForAnnotatedElement(
+            element: XElement,
+            prefixes: Map<XType, Array<String>>
+    ): List<String> {
+        return getAllDeeplinkUrIsFromCustomDeepLinksOnElement(
+                element = element,
+                prefixesMap = prefixes
+        ) + (element.getAnnotation(DEEP_LINK_CLASS)?.value?.value?.toList() ?: emptyList())
+    }
+
+    private fun verifyMethod(methodElement: XMethodElement) {
+        if (!methodElement.isStatic()) {
+            logError(
+                    element = methodElement,
+                    message = "Only static methods can be annotated with @${DEEP_LINK_CLASS.simpleName}",
+            )
+        }
+        // FIXME This is crashing with an NPE on internal classes when accessing the returnType.
```

Comment:
As far as I can tell this is a KSP bug

Reply:
@BenSchwab Yes, I added a ticket here: https://github.com/google/ksp/issues/492